yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+
+```bash
+-    #define BOOT_TARGET_DEVICES(func) \
+-    func(MMC, mmc, 0) \
++    #define BOOT_TARGET_DEVICES(func) \
++    func(MMC, mmc, 1) \
+```
+
+---
+# Build troubleshooting - bootloader
+
+* defconfig I
+
+.code-11px[```bash
+- CONFIG_SPL_MMC_SUPPORT=y
+- CONFIG_SPL_SERIAL_SUPPORT=y
++ CONFIG_SPL_MMC=y
++ CONFIG_SPL_SERIAL=y
+```]
+
+* defconfig II
+
+.code-11px[```bash
++ CONFIG_SYS_MALLOC_F_LEN=0x10000
++ CONFIG_SPL_SYS_MALLOC_F_LEN=0x2000
+```]
+
+.code-11px[```bash
+Before update:
+
+u-boot=> bdinfo
+(...)
+Early malloc usage: 1fe0 / 2000
+
+After update:
+
+alloc space exhausted
+(...)
+u-boot=> bdinfo
+(...)
+Early malloc usage: 1fe0 / 2000
+```]
+
+---
+# Build troubleshooting - kernel
+
+* Similar situation as with bootloader
+  - more often device-tree may take place in making problems, we are using more
+    peripherals from Linux side
+  - as in bootloader, we may face some of configurations goes from board config
+    into Kconfig files
+  - Yocto recipe compilation, but should not be a big issue as long as we are
+    able to compile it manually
+
+* defconfig
+
+.code-14px[```bash
++ CONFIG_EXPERT=y
++ CONFIG_GPIO_SYSFS=y
+```]
+
+???
+
+CONFIG_GPIO_SYSFS=y - enabling EXPERT in order to get the same config as before
+an update
+
+---
+# Build troubleshooting - kernel
+
+* device-tree I
+
+.code-14px[```bash
++	assigned-clocks = <&clk IMX8MM_CLK_PCIE1_AUX>,
++			  <&clk IMX8MM_CLK_PCIE1_PHY>,
++			  <&clk IMX8MM_CLK_PCIE1_CTRL>;
++	assigned-clock-rates = <10000000>, <100000000>, <250000000>;
++	assigned-clock-parents = <&clk IMX8MM_SYS_PLL2_50M>,
++				 <&clk IMX8MM_SYS_PLL2_100M>,
++				 <&clk IMX8MM_SYS_PLL2_250M>;
+```]
+
+* device-tree II
+
+.code-14px[```bash
+&sai2 {
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_sai2>;
+        pinctrl-1 = <&pinctrl_codec_en>;
+-       clocks = <&clk IMX8MM_CLK_SAI2_ROOT>;
+        assigned-clocks = <&clk IMX8MM_CLK_SAI2>;
+        assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
+        assigned-clock-rates = <24576000>;
+        (...)
+```]
+
+???
+
+Add `assigned-clocks`, `assigned-clock-rates` and
+`assigned-clock-parents` nodes to `pcie0`.
+
+mimics
+https://gitlab.com/3mdeb/dave_eu/linux-imx/-/commit/0d5920e4c35d15a9999ee19bb0521f21158072a0
+Specify the clock rate and parent of i.MX8MQ/MM PCIe clocks.
+
+sai2 overwriting correctly setting clocks in other file which caused whole
+platform hang while booting
+
+---
+# Build troubleshooting - default applications
+
+* Default applications as a part of created image
+  - by default we mean here part of systems to which we did not provide custom,
+    Client's source code
+  - issues here: configuration changes, compilation errors
+
+* Configuration changes
+  - missed or changed PACKAGECONFIG
+  - custom `do_configure` functions
+
+.code-14px[```bash
+- PACKAGECONFIG += "ssl libssh2"
++ PACKAGECONFIG += "openssl libssh2"
+```]
+
+tensorflow https://github.com/nxp-imx/meta-imx/pull/16
```

Comment:
Bullet missing? This issue also needs a reply from us

Reply:
moved to footnote, I want to reproduce that only with BSP release, will do before DevDay

https://github.com/3mdeb/conferences/pull/7/commits/e94ac909b1bc4d17c54aec915e9ef92e099235be#diff-a1666090e402e433c955cc46797dc399867480b4d815574efd9f5d4181e24ac3R563